discovery/src/main/scala/akka/discovery/SimpleServiceDiscovery.scala

```diff
@@ -41,13 +41,30 @@ object SimpleServiceDiscovery {
     }
   }
 
-  /** Resolved target host, with optional port and protocol spoken */
-  final case class ResolvedTarget(host: String, port: Option[Int]) {
+  /**
+   * Resolved target host, with optional port and the IP address.
+   * The address may be used for cluster bootstrap.
+   */
+  final case class ResolvedTarget(host: String, port: Option[Int], address: Option[String]) {
     def getPort: Optional[Int] = {
       import scala.compat.java8.OptionConverters._
       port.asJava
     }
 
+    def getAddress: Optional[String] = {
+      import scala.compat.java8.OptionConverters._
+      address.asJava
+    }
+
+    override def equals(o: Any): Boolean = o match {
+      case x: ResolvedTarget => (this.host == x.host) && (this.port == x.port)
+      case _ => false
+    }
```

Comment:
Hm... not entirely sure on this, port is part of the sorting after all, so lowest address depends on the port as well:

```
    implicit val addressOrdering: Ordering[ResolvedTarget] = Ordering.fromLessThan[ResolvedTarget] { (a, b) ⇒
      if (a eq b) false
      else if (a.host != b.host) a.host.compareTo(b.host) < 0
      else if (a.port != b.port) a.port.getOrElse(0) < b.port.getOrElse(0)
      else false
    }

```

Reply:
`port` is part of the comparison. `equals` is overridden specifically to exclude `address` out of the comparison.

Reply:
Hmm right, this should be fine

Reply:
Why is `address` specifically excluded? If two resolved target hosts for the same hostname resolve to different ip addresses, are they equal?